app/src/main/java/com/xinto/opencord/db/entity/channel/EntityLastMessageId.kt

```diff
@@ -0,0 +1,19 @@
+package com.xinto.opencord.db.entity.channel
+
+import androidx.compose.runtime.Immutable
+import androidx.room.ColumnInfo
+import androidx.room.Entity
+import androidx.room.PrimaryKey
+
+@Immutable
```

Comment:
Why is a DB entity marked Immutable?

Reply:
Doesn't seem very logical to me, you basically just merged entity, domain and api models into one?

Reply:
Even if they'd be identical, I think we should care more about consistency + code readability than saving a few bytes in memory

Reply:
Yes because they're identical
Well actually no not the api model

Reply:
Then why is the Entity marked Serializable 💀💀💀💀